src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -681,11 +748,11 @@ void func_80AF39DC(EnRu2* this, PlayState* play) {
     dialogState = Message_GetState(msgCtx);
 
     if (dialogState == TEXT_STATE_DONE_FADING) {
-        if (this->unk_2C3 != TEXT_STATE_DONE_FADING) {
+        if (this->lastDialogState != TEXT_STATE_DONE_FADING) {
             // "I'm Komatsu!" (cinema scene dev)
             PRINTF("おれが小松だ！ \n");
-            this->unk_2C2++;
-            if (this->unk_2C2 % 6 == 3) {
+            this->messageIndex++;
+            if (this->messageIndex % 6 == 3) {
                 player = GET_PLAYER(play);
```

Comment:
This doesnt seem like a "message index", its a timer that sets links position when it meets the condition of % 6 == 3

Reply:
What it does is switch cameras after enough advancements through Ruto's dialogue. There are two times the camera switches in the middle of the dialogue, and they are exactly six messages apart in the dialogue, hence being classified as a "message index."

Reply:
Maybe `textboxCounter` then or something. Message index to me sound like its setting a message id by using this value as an index to some array

Reply:
You may also consider adding a comment to the field
`/* 0x02C2 */ u8 textboxCount; // increements after every textbox is completed when talking on the first floor of water temple`

I think a few of these cutscene-specific variables should be clarified when they are relevant. For example, it doesnt seem like this variable counts up for EVERY textbox this actor handles, only the ones in this specific cutscene